rust/Earthfile

```
@@ -45,61 +53,60 @@ CARGO:
     DO +CHECK_INITED
     ARG --required args
     ARG output
-    ARG TMP_FOLDER="/tmp/earthly/lib/rust"
     IF [ "$EARTHLY_KEEP_FINGERPRINTS" = "false" ]
        DO +REMOVE_SOURCE_FINGERPRINTS
     END
-    DO +RUN_WITH_CACHE --command="set -e;
-        echo \"Running cargo $args\" ;
-        cargo $args;
-        if [ -n \"$output\" ]; then
-          echo \"Copying output files\" ;
-          mkdir -p $TMP_FOLDER;
-          cd target;
-          find . -type f -regextype posix-egrep -regex \"./$output\" -exec cp --parents \{\} $TMP_FOLDER \; ;
-          cd ..;
-        fi;
-        echo \"Running cargo sweep -r -t $EARTHLY_SWEEP_DAYS\" ;
-        cargo sweep -r -t $EARTHLY_SWEEP_DAYS;
-        echo \"Running cargo sweep -r -i\" ;
-        cargo sweep -r -i;"
+    DO +GET_RUST_CACHE_MOUNTS
+    RUN --mount=$EARTHLY_RUST_CARGO_HOME_CACHE --mount=$EARTHLY_RUST_TARGET_CACHE \
+      set -e; \
+      cargo $args; \
+      cargo sweep -r -t $EARTHLY_SWEEP_DAYS; \
+      cargo sweep -r -i;
     IF [ "$output" != "" ]
-      RUN mkdir -p target; \
-          mv $TMP_FOLDER/* target 2>/dev/null || echo "no files found within ./target matching the provided output regexp" ;
+      DO +COPY_OUTPUT --output=$output
     END
 
-# RUN_WITH_CACHE runs the passed command with the CARGO caches mounted.
-# Notice that in order to run this function, +INIT must be called first. This function exports the target cache mount ID under the env entry: $TARGET_CACHE_ID.
-# Arguments:
-#   - command (required): Command to run, can be any expression.
-#   - EARTHLY_CARGO_HOME_CACHE_ID: ID of the cargo home cache mount. By default: $EARTHLY_CARGO_HOME_CACHE_ID as exported by +INIT
-#   - target_cache_id: ID of the target cache mount. By default: ${EARTHLY_CARGO_HOME_CACHE_ID}#${EARTHLY_TARGET_NAME}
-#
-RUN_WITH_CACHE:
+# GET_RUST_CACHE_MOUNTS sets the following entries in the environment, to be used to mount the cargo caches.
+#   - EARTHLY_RUST_CARGO_HOME_CACHE: Code of the mount cache for the cargo home.
+#   - EARTHLY_RUST_TARGET_CACHE: Code of the mount cache for the target folder.
+# Notice that in order to run this function, +INIT must be called first.
+# Example:
+#   DO rust+RUST_GET_MOUNT_CACHE
+#   RUN --mount=$EARTHLY_RUST_CARGO_HOME_CACHE --mount=$EARTHLY_RUST_TARGET_CACHE cargo build --release
+GET_RUST_CACHE_MOUNTS:
```

I wonder if this should be called
```suggestion
SET_RUST_CACHE_MOUNTS:
```
?

---

Hi Ido, my perspective was that setting entries environment is a pattern to return values to the caller, something that might happen is the majority of functions, so the fact of setting the environment losses importance.

Instead, creating an entry of the choice of the function (like it is the case here) could be seeing as getting a value, while overwriting a entry given by the caller, setting it. 

Makes sense?

---

That is true, changed